### ISSUE TITLE
Add reflection tables

### ISSUE THREAD

**Issue by kaidaguerre** [turbot/steampipe#406]
## Reflection Tables
Steampipe will parse the `.sp` files (and other extensions for auto-resources, like `.sql`), and will add the definition information to a set of reflection tables.  These reflection tables will be added to the `internal` schema, and will allow users to query the mod configuration files using standard postgres syntax.  Note that these tables contain only the "definition" information as parsed; Outputs/results are not available as they are ephemeral.

- `steampipe_query`
- `steampipe_control`
- `steampipe_report`
- `steampipe_panel`
- `steampipe_variable`

***Are these a view on a `steampipe_resources` table?***

### Required columns for reflection tables
- mod_name
- resource_name
- file_name
- start_line_number
- end_line_number
- auto_generated (bool - true for `.sql` files, false for `.sp` files)
- source_definition 

Additional column for each argument
- commonly:
  - title
  - description
  - query
  - tags
  - sql

```sql
-- list all the controls in the aws mod
select * from steampipe_control where mod_name = 'aws'

-- list all the controls with a specific tag value
select * from steampipe_control where tag -> 'cis_version' = '1.3.0'

-- Show the hcl definition of a control
select mod_name, name, source_definition from steampipe_control where mod_name = 'aws' and resource_name = 'aws_cisv130_1_1'

-- show the sql for a query
select mod_name, name, sql from steampipe_query where mod_name = 'aws' and resource_name = 's3_bucket_encryption_in_transit_control'

```

***Is variable interpolation done in the columns? does the query return the statement, or 'query.my_query'?***

**Comment by kaidaguerre** [comment]
steampipe should populate these tables as part of the workspace loading

**Comment by kaidaguerre** [comment]
closed by 05254c3
